packages/create-plugin/src/commands/generate.command.ts

```diff
@@ -1,15 +1,318 @@
+import { getConfig } from '../utils/utils.config';
+import { getExportFileName } from '../utils/utils.files';
+import { normalizeId } from '../utils/utils.handlebars';
+import { getPackageManagerFromUserAgent, getPackageManagerInstallCmd } from '../utils/utils.packageManager';
+import { getExportPath } from '../utils/utils.path';
+import { getVersion } from '../utils/utils.version';
+import { TemplateData } from './types';
+import { renderTemplateFromFile } from '../utils/utils.templates';
+import { mkdir, writeFile } from 'node:fs/promises';
+import { updateGoSdkAndModules } from './generate-actions/update-go-sdk-and-packages';
+import { prettifyFiles } from './generate-actions/prettify-files';
+import { printGenerateSuccessMessage } from './generate-actions/print-success-message';
+import { printSuccessMessage } from '../utils/utils.console';
+
+const messages = {
+  generateFilesSuccess: `Successfully generated plugin files`,
+  updateGoSdkSuccess: `Successfully updated backend files`,
+  prettifySuccess: `Successfully prettified frontend files`,
+};
 
 export const generate = async (argv: minimist.ParsedArgs) => {
-  Plop.launch(
-    {
-      cwd: argv.cwd,
-      configPath: path.join(__dirname, 'generate.plopfile.js'),
-      require: argv.require,
-      completion: argv.completion,
-    },
-    (env) => run(env, undefined, true)
-  );
+  const answers = await promptUser(argv);
+  const templateData = getTemplateData(answers);
+  const exportPath = getExportPath(answers.pluginName, answers.orgName, answers.pluginType);
+  const templateActions = getTemplateActions({ templateData, exportPath });
+  const { pluginName, orgName, pluginType } = answers;
+  await generateFiles({ actions: templateActions, templateData });
+  printSuccessMessage(messages.generateFilesSuccess);
+  // @ts-ignore
+  await updateGoSdkAndModules({ pluginName, orgName, pluginType });
+  printSuccessMessage(messages.updateGoSdkSuccess);
+  // @ts-ignore
+  await prettifyFiles({ pluginName, orgName, pluginType });
+  printSuccessMessage(messages.prettifySuccess);
+  // @ts-ignore
+  console.log(printGenerateSuccessMessage(answers));
 };
+
+async function promptUser(argv: minimist.ParsedArgs) {
+  const answers: Record<string, any> = {};
+
+  for (const promptDefinition of prompts) {
+    const { name, type, message, validate, default: defaultValue, choices } = promptDefinition;
+    if (argv[name]) {
+      answers[name] = argv[name];
+    } else {
+      let prompt;
+
+      if (type === 'input') {
+        prompt = new Input({
+          name,
+          message,
+          validate,
+          initial: defaultValue,
+        });
+      }
+
+      if (type === 'select') {
+        prompt = new Select({
+          name,
+          message,
+          choices,
+        });
+      }
+
+      const promptResult = await prompt.run();
+      answers[name] = promptResult;
+    }
+  }
+
+  if (answers.pluginType !== PLUGIN_TYPES.panel) {
```

Comment:
We should aim to make this function more generic. That the questions can retrospect in previous answers would be the best. What if we allow an optional "shouldAsk" function on each prompt to which we pass the previous answers and tell us if a prompt should go ahead or not

Reply:
Thanks for the input. I think this is as generic as I can make it now. 👍